source/security.txt

```diff
@@ -13,13 +13,203 @@ Secure Your Data
 .. facet::
    :name: genre
    :values: reference
- 
+
 .. meta::
+   :keywords: ldap, authorize, ecs, aws, authenticate
    :description: Learn how to use {+driver-short+} to secure your data.
 
 .. toctree::
    :titlesonly:
    :maxdepth: 1
 
    /security/authentication
-   /security/enterprise-authentication
\ No newline at end of file
+   /security/enterprise-authentication
+
+Overview
+--------
+
+MongoDB supports multiple mechanisms that you can use to authenticate your application.
+This page contains code examples that demonstrate each of these mechanisms.
+
+.. tip::
+  
+  To learn more about any of the authentication mechanisms on this page, see the
+  :ref:`cpp-auth` and :ref:`cpp-enterprise-auth` pages.
+
+To use an authentication example from this page, copy the code example into the
+:ref:`sample application <cpp-auth-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<hostname>``, with
+the relevant values for your MongoDB deployment.
+
+.. _cpp-auth-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/connect-sample-app.cpp
```

Comment:
Recommend updating included code blocks to remove `#include` directives (not directly copy-pastable in the referenced example code region).

Suggest also updating the `client` variable declaration(s) to:

```cpp
auto client = mongocxx::client(uri);
```